### PR TITLE
Allow choosing a different VM size for the scheduler

### LINKED ISSUE #193: AzureVMCluster: scheduler takes entire VM
**What happened**:

Creating a Dask cluster of GPUs as in the example results in one VM being created for the scheduler, without also being available as a worker which is wasteful.

**What you expected to happen**:

I think this is a known issue, and I'm not sure what the solution is - I believe in AzureMLCluster we'd simply start the scheduler and worker on the same node.

**Minimal Complete Verifiable Example**:

```python
location="eastus"
resource_group = "dask-cloudprovider"
vnet = "dask-vnet"
security_group = "dask-nsg"

vm_size = "Standard_NC12s_v3"
docker_image="rapidsai/rapidsai:cuda11.0-runtime-ubuntu18.04-py3.8"
worker_class="dask_cuda.CUDAWorker"

from distributed import Client
from dask_cloudprovider.azure import AzureVMCluster

cluster = AzureVMCluster(location=location, resource_group=resource_group, vnet=vnet, security_group=security_group, vm_size=vm_size, docker_image=docker_image, worker_class=worker_class)
c = Client(cluster)
c
```

**Anything else we need to know?**:

Should this be fixed in the generic VMCluster? In each cloud provider? 

**Environment**:

- Dask version: 2.30.0
- Python version: 3.8
- Operating System: macOS
- Install method (conda, pip, source): pip

### PR DESCRIPTION
**Pull request by jacobtomlinson**
Closes #193.

@lostmygithubaccount would you mind testing this?

### CODE REVIEW

**Comment by lostmygithubaccount** [comment]
is there any reason for developing on forks instead of branches for Dask?

**Comment by mrocklin** [comment]
It's hard to give all possible contributors commit access

**Comment by lostmygithubaccount** [comment]
I'm hitting an error - @jacobtomlinson any ideas? for context I:

- create a resource group named `dask-cloudprovider`
- create `dask-vnet` and `dask-nsg`
- create a few dask clusters and spun them down
- a bunch of NIC resources were left around after the VMs were cleaned up, I manually deleted them
- tried recreating dask clusters, hitting this

image

I've tried new resource groups.

**Comment by jacobtomlinson** [comment]
Oh dear. Looks like you aren't allowed to create NICs with public IP address maybe?

**Comment by lostmygithubaccount** [comment]
but I definitely was able to the other day...I've tried getting around this by setting `public_ingress=False` and it seems stuck waiting for the scheduler to run

**Comment by lostmygithubaccount** [comment]
seems like this is an unrelated internal subscription policy issue - blocked on testing by this, but the changes look good other than `vm_size` is repeated twice in the docstring instead of `scheduler_vm_size`

File: dask_cloudprovider/azure/azurevm.py [hunk 1]
```diff
@@ -226,6 +226,8 @@ class AzureVMCluster(VMCluster):
     vm_size: str
         Azure VM size to use for scheduler and workers. Default ``Standard_DS1_v2``.
         List available VM sizes with ``az vm list-sizes --location <location>``.
+    vm_size: str
```
[the remark this answers is not in the document]
Comment (jacobtomlinson) [reply dask_cloudprovider/azure/azurevm.py:229]:
```suggestion
    scheduler_vm_size: str
```

**Comment by jacobtomlinson** [comment]
Thanks @lostmygithubaccount

**Pull request merged**